engine/table/src/main/java/io/deephaven/engine/table/impl/by/PartitionByChunkedOperator.java

```diff
@@ -477,19 +470,46 @@ private boolean appendShifts(@NotNull final LongChunk<? extends RowKeys> preShif
         }
         if (builder == null) {
             final RowSet tableRowSet = tables.getUnsafe(destination).getRowSet();
-            final RowSetBuilderRandom removedRowSetBuilder = removedBuilders.getUnsafe(destination);
+            final RowSetBuilderRandom removedRowSetBuilder =
+                    (RowSetBuilderRandom) removedBuilders.getUnsafe(destination);
             final RowSet preShiftKeys;
             if (removedRowSetBuilder == null) {
                 preShiftKeys = tableRowSet.copy();
             } else {
-                // we want to keep the builder in `removedRowSetBuilders` ready for a call to `extractAndClearRowSet` so
-                // we remake the builder after this call
-                try (final RowSet rs = removedRowSetBuilder.build()) {
-                    preShiftKeys = tableRowSet.minus(rs);
-                    RowSetBuilderRandom newBuilder = RowSetFactory.builderRandom();
-                    newBuilder.addRowSet(rs);
-                    removedBuilders.set(destination, newBuilder);
+                // dummy builder to allow a subsequent call to build()`
+                class PrecomputedRowSetBuilder implements RowSetBuilderRandom {
+                    WritableRowSet rowSet;
+
+                    public PrecomputedRowSetBuilder(final WritableRowSet rowSet) {
+                        this.rowSet = rowSet;
+                    }
+
+                    @Override
+                    public WritableRowSet build() {
+                        // make sure to release our reference to the rowset after `build()` is called
+                        WritableRowSet tmp = rowSet;
+                        rowSet = null;
+                        return tmp;
+                    }
+
+                    @Override
+                    public void addKey(long rowKey) {
+                        throw (new UnsupportedOperationException("PrecomputedRowSetBuilder cannot be modified)"));
```

Comment:
Why the weird parentheses here?

Reply:
TIL (Today I Learned) that `throw` is a keyword and not a function call